src/backend/distributed/cdc/cdc_decoder.c

```diff
@@ -78,8 +74,8 @@ _PG_output_plugin_init(OutputPluginCallbacks *cb)
 	elog(LOG, "Initializing CDC decoder");
 	LogicalOutputPluginInit plugin_init =
 		(LogicalOutputPluginInit) (void *)
-		load_external_function(CDC_BASE_DECODER_PLUGIN_NAME,
-							   CDC_BASE_DECODER_PLUGIN_INIT_FUNCTION_NAME,
+		load_external_function(DECODER_NAME,
```

Comment:
If the user configures dynamic_library_path to include the citus_decoders/ directory, such that "pgoutput" points to our binary, then load_external_funcction would recurse back into this function.

To be able to make the citus_decoders/ trick work, we need to temporarily remove the entry containing citus_decoders from the Dynamic_library_path here.

Reply:
see find_in_dynamic_libpath for parsing logic

Reply:
Ok will do this.